test/Conversion/ExportVerilog/max-terms.mlir

```diff
@@ -25,13 +26,15 @@ hw.module @large_use_in_procedural(%clock: i1, %a: i1) {
   }
 
   %reg = sv.reg : !hw.inout<i1>
+
+  // CHECK: assign [[GEN_0]] = reg_0 + reg_0 + reg_0 + reg_0 + reg_0;
   sv.alwaysff(posedge %clock) {
-    // CHECK: [[REG]] <= a;
-    sv.passign %reg, %a : i1
+    // CHECK: always
+    // CHECK: [[REG]] = a;
+    sv.bpassign %reg, %a : i1
     %0 = sv.read_inout %reg : !hw.inout<i1>
-    // This expression cannot be hoisted, even though it's over the limit.
```

Comment:
I think this looks like broken, but it is a rather bug of `hoistNonSideEffectExpr` that hoist `sv.read_inout` (related issue https://github.com/llvm/circt/issues/3028).